scripts/ws-k8s-guard.sh

```
@@ -155,5 +215,46 @@ k8s_guard_evaluate() {
+    # keeps the whole remainder as the reason and uses the scope remediation.
+    case "$class" in
+        scope|unbounded|precondition|context) ;;
+        *) reason="$body"; class="scope" ;;
+    esac
+    printf 'REJECTED by the k8s scope guard: %s.' "$reason"
+    case "$class" in
+        scope)
+            # The target namespace is the thing out of scope — adding it (whether
+            # for a pod write or a create/delete of that namespace) authorizes the
+            # op for just that namespace. Avoid the vague "widen the scope".
+            printf ' Reads are free cluster-wide. Add that namespace to the scope to authorize this for just that namespace (`ws k8s scope set --context %s --namespace <ns,...>`), or run plain `kubectl` outside the guard.' "$ctx" ;;
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)